zaza/openstack/charm_tests/heat/tests.py

```diff
@@ -0,0 +1,311 @@
+import zaza.openstack.charm_tests.test_utils as test_utils
+import zaza.openstack.utilities.openstack as openstack_utils
+import zaza.charm_lifecycle.utils as charm_lifecycle_utils
+
+# Resource and name constants
+IMAGE_NAME = 'cirros'
+STACK_NAME = 'hello_world'
+RESOURCE_TYPE = 'server'
+TEMPLATES_PATH = 'files'
+FLAVOR_NAME = 'm1.tiny'
+
+
+class HeatBasicDeployment(test_utils.OpenStackBaseTest):
+    """Encapsulate Heat tests."""
+
+    @classmethod
+    def setUpClass(cls):
+        """Run class setup for running Heat tests."""
+        super(HeatBasicDeployment, cls).setUpClass()
+        cls.application = 'heat'
+        cls.keystone_session = openstack_utils.get_overcloud_keystone_session()
+        cls.heat_client = openstack_utils.get_heat_session_client(
+            cls.keystone_session)
+        cls.glance_client = openstack_utils.get_glance_session_client(
+            cls.keystone_session)
+        cls.nova_client = openstack_utils.get_nova_session_client(
+            cls.keystone_session)
+
+    @property
+    def services(self):
+        """Return a list services for OpenStack release.
+
+        :returns: List of services
+        :rtype: [str]
+        """
+        services = ['heat-api', 'heat-api-cfn', 'heat-engine']
+        return services
+
+    def test_100_domain_setup(self):
+        """Run required action for a working Heat unit."""
+        # Action is REQUIRED to run for a functioning heat deployment
+        logging.info('Running domain-setup action on heat unit...')
+        unit = zaza.model.get_units(self.application_name)[0]
+        zaza.model.block_until_unit_wl_status(unit.entity_id, "active")
+        zaza.model.run_action(unit.entity_id, "domain-setup")
+        zaza.model.block_until_unit_wl_status(unit.entity_id, "active")
+
+    def test_400_heat_resource_types_list(self):
+        """Check default resource list behavior and confirm functionality."""
+        logging.info('Checking default heat resource list...')
+        try:
+            types = self.heat_client.resource_types.list()
+            if type(types) is list:
+                logging.info('Resource type list check is ok.')
+            else:
+                msg = 'Resource type list is not a list!'
+                assert False, msg
+            if len(types) > 0:
+                logging.info('Resource type list is populated '
+                             '({}, ok).'.format(len(types)))
+            else:
+                msg = 'Resource type list length is zero!'
+                assert False, msg
+        except Exception as e:
+            msg = 'Resource type list failed: {}'.format(e)
+            assert False, msg
+
+    def test_410_heat_stack_create_delete(self):
+        """Create stack, confirm nova compute resource, delete stack."""
+        # Create an image for use by the heat template
+        logging.info('Creating glance image ({})...'.format(IMAGE_NAME))
+        glance_setup.add_cirros_image(image_name=IMAGE_NAME)
```

Comment:
Instead of calling the helper here, use a `configure` line in the tests.yaml of the test.  i.e.

```yaml
configure:
  - zaza.charm_tests.glance.setup.add_cirros_image
```